lighthouse-core/gather/connections/connection.js

```diff
@@ -26,6 +26,27 @@ class Connection {
     /** @type {!Map<number, {resolve: function(*), reject: function(*), method: string}>}*/
     this._callbacks = new Map();
     this._eventEmitter = new EventEmitter();
+    this._messageLog = [];
+    this._recordMessages = false;
+  }
+
+  /**
+   * @return {!Array<Object>}
+   */
+  get messageLog() {
```

Comment:
maybe bump this stuff into it's own class and instantiate it in the connection constructor?